ui/apps/platform/src/Containers/Collections/CollectionAttacher.tsx

```diff
@@ -1,7 +1,95 @@
+import useEmbeddedCollections from './hooks/useEmbeddedCollections';
+import { CollectionSlim } from './types';
+
+const selectorListCells = [
+    {
+        name: 'Name',
+        render: ({ name }) => (
+            <Button variant="link" className="pf-u-pl-0">
+                {name}
+            </Button>
+        ),
+    },
+    {
+        name: 'Description',
+        render: ({ description }) => <Truncate content={description} />,
+    },
+];
+
+export type CollectionAttacherProps = {
+    initialEmbeddedCollections: CollectionSlim[];
+    onSelectionChange: (collections: CollectionSlim[]) => void;
+};
+
+function compareNameLowercase(search: string): (item: { name: string }) => boolean {
+    return ({ name }) => name.toLowerCase().includes(search.toLowerCase());
+}
+
+function CollectionAttacher({
+    initialEmbeddedCollections,
+    onSelectionChange,
+}: CollectionAttacherProps) {
+    const [search, setSearch] = useState('');
+    const embedded = useEmbeddedCollections(initialEmbeddedCollections);
+    const { attached, detached, attach, detach, hasMore, fetchMore, onSearch } = embedded;
+    const { fetchMoreLoading, fetchMoreError } = embedded;
+
+    const onSearchInputChange = useMemo(
+        () =>
+            debounce((value: string) => {
+                setSearch(value);
+                onSearch(value);
+            }, 800),
+        [onSearch]
+    );
+
+    const selectedOptions = attached.filter(compareNameLowercase(search));
+    const deselectedOptions = detached.filter(compareNameLowercase(search));
+
+    return (
+        <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsXl' }}>
+            <SearchInput
+                aria-label="Search by name"
+                placeholder="Search by name"
+                value={search}
+                onChange={onSearchInputChange}
+            />
+            <BacklogListSelector
+                selectedOptions={selectedOptions}
+                deselectedOptions={deselectedOptions}
+                onSelectItem={({ id }) => attach(id)}
+                onDeselectItem={({ id }) => detach(id)}
+                onSelectionChange={onSelectionChange}
+                rowKey={({ id }) => id}
+                cells={selectorListCells}
+                selectedLabel="Attached collections"
+                deselectedLabel="Detached collections"
+                selectButtonText="Attach"
+                deselectButtonText="Detach"
+            />
+            {fetchMoreError && (
+                <Alert
+                    variant="danger"
+                    isInline
+                    title="There was an error loading more collections"
+                />
+            )}
+            {hasMore && (
+                <Button
+                    className="pf-u-align-self-flex-start"
+                    variant="secondary"
+                    onClick={() => fetchMore(search)}
+                    isLoading={fetchMoreLoading}
```

Comment:
Most boolean state variables have `is` or occasionally `has` prefix:

```js
isLoading={isFetchMoreLoading}
```

An grammatical alternative in this case which I mention, not necessarily recommend: `isFetchingMore`

Reply:
I think the grammatical alternative reads better in this case, updated the prop name.